packages/nx/src/command-line/release/release.ts

```diff
@@ -0,0 +1,110 @@
+import { prompt } from 'enquirer';
+import { readNxJson } from '../../config/nx-json';
+import { output } from '../../devkit-exports';
+import { createProjectGraphAsync } from '../../project-graph/project-graph';
+import { handleErrors } from '../../utils/params';
+import { releaseChangelog } from './changelog';
+import { ReleaseOptions, VersionOptions } from './command-object';
+import {
+  createNxReleaseConfig,
+  handleNxReleaseConfigError,
+} from './config/config';
+import { releasePublish } from './publish';
+import { resolveNxJsonConfigErrorMessage } from './utils/resolve-nx-json-error-message';
+import { NxReleaseVersionResult, releaseVersion } from './version';
+
+export const releaseCLIHandler = (args: VersionOptions) =>
+  handleErrors(args.verbose, () => release(args));
+
+export async function release(
+  args: ReleaseOptions
+): Promise<NxReleaseVersionResult | number> {
+  const projectGraph = await createProjectGraphAsync({ exitOnError: true });
+  const nxJson = readNxJson();
+
+  if (args.verbose) {
+    process.env.NX_VERBOSE_LOGGING = 'true';
+  }
+
+  if (
```

Comment:
I think this is unnecessarily complex and granular. We can check for `Object.keys(version.git).length` and have the `jsonConfigErrorPath` be `version.git` (and same for changelog)

Reply:
```ts
const hasVersionGitConfig =
  Object.keys(nxJson.release?.version?.git ?? {}).length > 0;
const hasChangeLogGitConfig =
  Object.keys(nxJson.release?.changelog?.git ?? {}).length > 0;

if (hasVersionGitConfig || hasChangeLogGitConfig) {
  const jsonConfigErrorPath = hasVersionGitConfig
    ? ["release", "version", "git"]
    : ["release", "changelog", "git"];
}
```